endpoints/resources/src/main.cpp

```diff
@@ -221,49 +222,133 @@ namespace
+		                                       _args = std::move(_args)] {
+			logging::info("{}: client_info.username = [{}]", fn, client_info.username);
+
+			http::response<http::string_body> res{http::status::ok, _req.version()};
+			res.set(http::field::server, irods::http::version::server_name);
+			res.set(http::field::content_type, "application/json");
+			res.keep_alive(_req.keep_alive());
+
+			try {
+				const auto name_iter = _args.find("name");
+				if (name_iter == std::end(_args)) {
+					logging::error("{}: Missing [name] parameter.", fn);
+					return _sess_ptr->send(irods::http::fail(http::status::bad_request));
+				}
+
+				const auto property_iter = _args.find("property");
+				if (property_iter == std::end(_args)) {
+					logging::error("{}: Missing [property] parameter.", fn);
+					return _sess_ptr->send(irods::http::fail(http::status::bad_request));
+				}
+
+				const auto value_iter = _args.find("value");
+				if (value_iter == std::end(_args)) {
+					logging::error("{}: Missing [value] parameter.", fn);
+					return _sess_ptr->send(irods::http::fail(http::status::bad_request));
+				}
+
+				auto conn = irods::get_connection(client_info.username);
+
+				if (property_iter->second == "name") {
+					// TODO(???): Remove this once the resource administration library grows support
+					// for renaming resources.
+					//
+					// The following is equivalent to what the resource administration library would
+					// provide.
+					GeneralAdminInput input{};
+					input.arg0 = "modify";
+					input.arg1 = "resource";
+					input.arg2 = name_iter->second.c_str();
+					input.arg3 = "name";
+					input.arg4 = value_iter->second.c_str();
+
+					if (const auto ec = rcGeneralAdmin(static_cast<RcComm*>(conn), &input); ec < 0) {
+						constexpr const char* msg = "Could not change name from [{}] to [{}] for resource [{}]";
+						// NOLINTNEXTLINE(cppcoreguidelines-pro-bounds-array-to-pointer-decay)
+						THROW(ec, fmt::format(msg, input.arg3, input.arg4, input.arg2));
+					}
+				}
+				else if (property_iter->second == "type") {
+					adm::client::modify_resource(
+						conn, name_iter->second, adm::resource_type_property{value_iter->second});
+				}
+				else if (property_iter->second == "host") {
+					adm::client::modify_resource(conn, name_iter->second, adm::host_name_property{value_iter->second});
+				}
+				else if (property_iter->second == "vault_path") {
+					adm::client::modify_resource(conn, name_iter->second, adm::vault_path_property{value_iter->second});
+				}
+				else if (property_iter->second == "status") {
+					auto status_enum = adm::resource_status::up;
+
+					if (value_iter->second == "up") {
+						status_enum = adm::resource_status::up;
+					}
+					else if (value_iter->second == "down") {
+						status_enum = adm::resource_status::down;
+					}
+					else {
+						logging::error(
+							"{}: Invalid value for [value] parameter. Received [{}].", fn, value_iter->second);
```

Comment:
Should we state the name of the property ("status") in the message as well?

Reply:
Will adjust.

Reply:
Done.